service/src/main/java/uk/gov/hmcts/reform/fpl/config/feign/errordecoder/IdamErrorDecoder.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.reform.fpl.config.feign.errordecoder;
+
+import feign.Response;
+import feign.RetryableException;
+import feign.codec.ErrorDecoder;
+import org.springframework.http.HttpStatus;
+
+public class IdamErrorDecoder implements ErrorDecoder {
+
+    private final ErrorDecoder defaultDecoder;
+
+    public IdamErrorDecoder() {
+        this(new Default());
+    }
+
+    public IdamErrorDecoder(ErrorDecoder defaultDecoder) {
+        this.defaultDecoder = defaultDecoder;
+    }
+
+    @Override
+    public Exception decode(String methodKey, Response response) {
+        Exception exception = defaultDecoder.decode(methodKey, response);
+
+        if (exception instanceof RetryableException) {
+            return exception;
+        }
+
+        if (HttpStatus.valueOf(response.status()).is5xxServerError()) {
+            return new RetryableException(
+                response.status(),
+                response.reason(),
+                response.request().httpMethod(),
+                exception,
+                null,
+                response.request()
+            );
+        }
+
+        return null;
```

Comment:
are you sure about this one? Default decode returns exception from line 22

Reply:
yeah, my bad would want to return the original exception as otherwise we would be getting NPEs in the code